### PR TITLE
Backward forward

### PR DESCRIPTION
**Pull request by Saelyos**
Add an experimental backward forward solver.

### CODE REVIEW

File: roseau/load_flow/_solvers.py [hunk 1]
```diff
@@ -225,3 +228,41 @@ def update_params(self, params: JsonDict) -> None:
...
+                The electrical network for which the load flow needs to be solved.
+        """
+        super().__init__(network=network)
+        self._cy_solver = CyBackwardForward(network=network._cy_electrical_network)
+
+    def update_network(self, network: "ElectricalNetwork") -> None:
+        self._cy_solver = CyBackwardForward(network=network._cy_electrical_network)
+
+    def update_params(self, params: JsonDict) -> None:
+        pass
```
Comment (benoit9126) [inline roseau/load_flow/_solvers.py:254]: I would have raised an error or emitting a warning to ensure the user understands that something is weird about his code

**Review by benoit9126 — APPROVED** [review]

File: doc/advanced/Solvers.md [hunk 2]
```diff
@@ -131,3 +131,30 @@ The _Goldstein and Price_ solver accepts the following parameters:
...
+### Advantages
+
+- **Faster Execution Time**: The backward-forward sweep method generally executes more quickly than Newton-Raphson-based
+  approaches.
+
+### Limitations
+
+- **Modeling Restrictions**:
+  - The network cannot contain loops.
+  - Floating neutrals are not allowed for loads.
```
Comment (alihamdan) [inline doc/advanced/Solvers.md:149]:
```suggestion
  - Floating neutrals are not allowed for loads and sources.
```

File: doc/advanced/Solvers.md [hunk 3]
```diff
@@ -131,3 +131,30 @@ The _Goldstein and Price_ solver accepts the following parameters:
...
+- **Convergence**: This method may exhibit weaker convergence properties, particularly in scenarios involving
+  flexible loads.
+
+### Important Note
+
+This solver is still in the experimental stage. Users should be aware that certain edge cases may not be handled
+correctly when utilizing the backward-forward sweep method.
```
Comment (alihamdan) [inline doc/advanced/Solvers.md:156]:
```suggestion
This solver is still in the **experimental** stage. Users should be aware that certain edge cases may not be handled
correctly when utilizing the backward-forward sweep method. If you face any issues with this solver, please let
us know by filing a GitHub issue.
```

**Review by alihamdan — APPROVED** [review]
Thanks, just a couple minor documentation suggestions.

**Pull request merged**